File: app/src/main/java/ru/mpei/vmss/myapplication/fragments/Tasks.java
package ru.mpei.vmss.myapplication.fragments;

import android.content.Context;
import android.os.Build;
import android.os.Bundle;

import androidx.annotation.RequiresApi;
import androidx.fragment.app.Fragment;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ExpandableListView;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import ru.mpei.vmss.myapplication.R;
import ru.mpei.vmss.myapplication.adapters.TasksAdapter;
import ru.mpei.vmss.myapplication.pojo.Task;

import static ru.mpei.vmss.myapplication.fragments.User.isAuth;
import static ru.mpei.vmss.myapplication.fragments.User.userId;

public class Tasks extends Fragment {

    private Context context;
    private TasksAdapter adapter;
    private List<Task> dataList = new ArrayList<>();

    public Tasks() {
    }

    @RequiresApi(api = Build.VERSION_CODES.M)
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        context = Objects.requireNonNull(getActivity()).getApplicationContext();

        View rootView = inflater.inflate(R.layout.fragment_tasks, container, false);
        ExpandableListView list = rootView.findViewById(R.id.tasksList);

        adapter = new TasksAdapter(context, dataList, 0);
        list.setAdapter(adapter);
        list.setOnGroupExpandListener(groupPosition -> {
            for (int g = 0; g < adapter.getGroupCount(); g++) {
                if (g != groupPosition) { list.collapseGroup(g);
                }
            }
        });
        updateList();

        SwipeRefreshLayout refresher = rootView.findViewById(R.id.tasksRefresher);
        refresher.setColorSchemeColors(context.getColor(R.color.bgBottomNavigation));
        refresher.setOnRefreshListener(() -> {
            updateList();
            refresher.setRefreshing(false);
        });

        return rootView;
    }

    public void updateList() {
        dataList.clear();

        JSONObject body = new JSONObject();
        try {
            if(isAuth) {
                body.put("id", userId);
            } else {
                body.put("id", 0);
            }
            body.put("type", "available");
        } catch (JSONException e) {
            e.printStackTrace();
        }

        JsonArrayRequest request = new JsonArrayRequest(Request.Method.POST,
                context.getString(R.string.getTasksUrl), body, response -> {
                    for (int i = 0; i < response.length(); i++){
                        JSONObject obj;
                        try {
                            obj = response.getJSONObject(i);
                            dataList.add( new Task(obj.getString("description_short"),
                                    obj.getString("description_full"),
                                    obj.getString("place"),
                                    obj.getString("reward"),
                                    obj.getString("start"),
                                    obj.getString("end"),
                                    obj.getString("refuse_before"),
                                    obj.getString("id")));

                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }
                    adapter.notifyDataSetChanged();
                },
                error -> Toast.makeText(context, "Ошибка соединения", Toast.LENGTH_LONG).show());

        RequestQueue requestQueue = Volley.newRequestQueue(context);
        requestQueue.add(request);
    }

}

File: app/src/main/java/ru/mpei/vmss/myapplication/fragments/Tasks.kt
package ru.mpei.vmss.myapplication.fragments

import android.os.Build
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.fragment.app.Fragment
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.VolleyError
import com.android.volley.toolbox.JsonArrayRequest
import com.android.volley.toolbox.Volley
import kotlinx.android.synthetic.main.fragment_tasks.*
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import ru.mpei.vmss.myapplication.R
import ru.mpei.vmss.myapplication.adapters.TasksAdapter
import ru.mpei.vmss.myapplication.pojo.Task
import java.util.*

class Tasks : Fragment() {
    private var adapter: TasksAdapter? = null
    private val dataList: MutableList<Task> = ArrayList()

    @RequiresApi(api = Build.VERSION_CODES.M)
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_tasks, container, false)
    }


    @RequiresApi(Build.VERSION_CODES.M)
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        adapter = TasksAdapter(requireContext(), dataList)
        tasksList.setAdapter(adapter)
        tasksList.setOnGroupExpandListener { groupPosition: Int ->
            for (g in 0 until adapter!!.groupCount) {
                if (g != groupPosition) {
                    tasksList.collapseGroup(g)
                }
            }
        }
        updateList()
        tasksRefresher.setColorSchemeColors(requireContext().getColor(R.color.bgBottomNavigation))
        tasksRefresher.setOnRefreshListener {
            updateList()
            tasksRefresher.isRefreshing = false
        }
    }

    fun updateList() {
        dataList.clear()
        val body = JSONObject()
        try {
            if (User.isAuth) {
                body.put("id", User.userId)
            } else {
                body.put("id", 0)
            }
            body.put("type", "available")
        } catch (e: JSONException) {
            e.printStackTrace()
        }
        val request = JsonArrayRequest(Request.Method.POST,
                requireContext().getString(R.string.getTasksUrl), body, Response.Listener { response: JSONArray ->
            for (i in 0 until response.length()) {
                var obj: JSONObject
                try {
                    obj = response.getJSONObject(i)
                    dataList.add(Task(obj.getString("description_short"),
                            obj.getString("description_full"),
                            obj.getString("place"),
                            obj.getString("reward"),
                            obj.getString("start"),
                            obj.getString("end"),
                            obj.getString("refuse_before"),
                            obj.getString("id")))
                } catch (e: JSONException) {
                    e.printStackTrace()
                }
            }
            adapter!!.notifyDataSetChanged()
        },
                Response.ErrorListener { Toast.makeText(requireContext(), "Ошибка соединения", Toast.LENGTH_LONG).show() })
        val requestQueue = Volley.newRequestQueue(requireContext())
        requestQueue.add(request)
    }
}